packages/transactions/src/pc.ts

```diff
@@ -0,0 +1,229 @@
+   * A post-condition sending `FungibleConditionCode.LessEqual` uSTX or other fungible tokens.
+   * Finalize with the chained `.ustx()` or `.ft(…)` method.
+   */
+  willSendLte(amount: IntegerType) {
+    return new PartialPcFtWithCode(
+      this.address,
+      amount,
+      FungibleConditionCode.LessEqual,
+      this.contractName
+    );
+  }
+
+  /**
+   * ### Fungible Token Post Condition
+   * A post-condition sending `FungibleConditionCode.Less` uSTX or other fungible tokens.
+   * Finalize with the chained `.ustx()` or `.ft(…)` method.
+   */
+  willSendLt(amount: IntegerType) {
+    return new PartialPcFtWithCode(
+      this.address,
+      amount,
+      FungibleConditionCode.Less,
+      this.contractName
+    );
+  }
+
+  /**
+   * ### Fungible Token Post Condition
+   * A post-condition sending `FungibleConditionCode.GreaterEqual` uSTX or other fungible tokens.
+   * Finalize with the chained `.ustx()` or `.ft(…)` method.
+   */
+  willSendGte(amount: IntegerType) {
+    return new PartialPcFtWithCode(
+      this.address,
+      amount,
+      FungibleConditionCode.GreaterEqual,
+      this.contractName
+    );
+  }
+
+  /**
+   * ### Fungible Token Post Condition
+   * A post-condition sending `FungibleConditionCode.Greater` uSTX or other fungible tokens.
+   * Finalize with the chained `.ustx()` or `.ft(…)` method.
+   */
+  willSendGt(amount: IntegerType) {
+    return new PartialPcFtWithCode(
+      this.address,
+      amount,
+      FungibleConditionCode.Greater,
+      this.contractName
+    );
+  }
+
+  /**
+   * ### NFT Post Condition
+   * A post-condition which `NonFungibleConditionCode.Sends` an NFT.
+   * Finalize with the chained `.nft(…)` method.
+   */
+  willSendAsset() {
+    return new PartialPcNftWithCode(
+      this.address,
+      NonFungibleConditionCode.Sends,
+      this.contractName
+    );
+  }
+
+  /**
+   * ### NFT Post Condition
+   * A post-condition which `NonFungibleConditionCode.DoesNotSend` an NFT.
+   * Finalize with the chained `.nft(…)` method.
+   */
+  willNotSendAsset() {
+    return new PartialPcNftWithCode(
+      this.address,
+      NonFungibleConditionCode.DoesNotSend,
+      this.contractName
+    );
+  }
+}
+
+class PartialPcFtWithCode {
+  constructor(
+    private address: string,
+    private amount: IntegerType,
+    private code: FungibleConditionCode,
+    private contractName?: string
+  ) {}
+
+  /**
+   * ### STX Post Condition
+   * ⚠ Amount of STX is denoted in uSTX (micro-STX)
+   */
+  ustx() {
+    // todo: rename to `uSTX`?
+    if (this.contractName) {
+      return makeContractSTXPostCondition(this.address, this.contractName, this.code, this.amount);
+    }
+    return makeStandardSTXPostCondition(this.address, this.code, this.amount);
+  }
+
+  /**
+   * ### Fungible Token Post Condition
+   * ⚠ Amount of fungible tokens is denoted in the smallest unit of the token
+   */
+  ft(contractId: ContractIdString, tokenName: string) {
```

Comment:
For this and also nfts, is it common to use a fully-qualified identifier in the format `<principal>::<token_name>`? If so, might be useful to also support that format as an arg?

Reply:
Oh nice, good to know -- will add that